### PR TITLE
fix(libs/utils): rely on DNS for rpc/grpc requests to core

### PR DESCRIPTION
**Pull request by ramin**
Refs #3570 and I believe solves the issue well enough

previously, on start, node would resolve DNS hostnames to IP when validating. This is somewhat legitimate and necessary when building multiaddr format, but the same code was being used for configuring http clients for RPC/gRPC connections. 

This meant, that in the event of some server administration or an incident, if DNS was updated for a "core" node (as configured via `--core.ip`) this meant a long running node would suddenly lose it's destination and get stuck failing and retrying. 

Instead, this seeks to separate validation, for validating --core.ip for rpc/grpc clients do NOT resolve to IP on start/validate, then should make requests over the DNS using hostname, meaning a core node moving, should eventually resolve again properly, preventing need for a restart.

We had to take care NOT to accidentally cause multiaddr to remain hostnames. The original issue suggests wanting to update all p2p networking to be host based, I feel that is probably unnecessary

### CODE REVIEW

File: libs/utils/address.go [hunk 1]
```diff
@@ -37,9 +52,5 @@ func ValidateAddr(addr string) (string, error) {
 		return addr, nil
 	}
 
-	resolved, err := net.ResolveIPAddr("ip4", addr)
-	if err != nil {
-		return addr, err
-	}
-	return resolved.String(), nil
+	return addr, nil
```
Comment (cristaloleg) [inline libs/utils/address.go:55]: Looks like this and `return` above are identical. Probably we should return error in `return` above.

Also, we can use https://pkg.go.dev/net/netip#ParseAddr to get error message, `net.ParseIP` is an old and not friendly way to parse ip.

File: libs/utils/address.go [hunk 2]
```diff
@@ -24,9 +24,24 @@ func SanitizeAddr(addr string) (string, error) {
...
-// sanitized address is returned.
+// ValidateAddr sanitizes the given address and verifies that it
+// is a valid IP or hostname. The sanitized address is returned.
 func ValidateAddr(addr string) (string, error) {
+	addr, err := ValidateAddrPreservingHostname(addr)
+	if err != nil {
+		return addr, err
+	}
+
+	resolved, err := net.ResolveIPAddr("ip4", addr)
```
Comment (cristaloleg) [inline libs/utils/address.go:35]: Also can be https://pkg.go.dev/net/netip#ParseAddr

**Review by renaynay — COMMENTED** [review]
agree with @cristaloleg comments

**Review by cristaloleg — APPROVED** [review]
LGTM, thanks!

**Review by Wondertan — APPROVED** [review]
Agree with p2p networking reasoning. Thanks@

**Pull request merged**